### DISCUSSION TITLE
Feedback on New GitHub Code View

### DISCUSSION THREAD

**Discussion by AliAryanTech** [community/community#55095] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

Dear GitHub Team,

I'm a phone user
I wanted to provide some feedback on the recent changes made to the GitHub code view. While I appreciate the updates, I have encountered a few issues that have affected my coding experience negatively.

The new code view is laggy, glitchy, and makes it hard to delete new lines or copy-paste code. Personally, I preferred the previous version as it was smoother and more user-friendly.

Please consider the feedback from developers like myself who are facing difficulties with the new code view. Enhancing the user experience will help maintain GitHub as a beloved platform for developers worldwide.

[![Well](https://telegra.ph/file/32681b28f0c194a796edc.png)](https://telegra.ph/file/fa078713c5e6236075f3b.mp4)

**Comment by Billywildhig** [comment]
> ### Select Topic Area
> 
> Product Feedback
> 
> ### Body
> 
> Dear GitHub Team,
> 
> 
> 
> I'm a phone user
> 
> I wanted to provide some feedback on the recent changes made to the GitHub code view. While I appreciate the updates, I have encountered a few issues that have affected my coding experience negatively.
> 
> 
> 
> The new code view is laggy, glitchy, and makes it hard to delete new lines or copy-paste code. Personally, I preferred the previous version as it was smoother and more user-friendly.
> 
> 
> 
> Please consider the feedback from developers like myself who are facing difficulties with the new code view. Enhancing the user experience will help maintain GitHub as a beloved platform for developers worldwide.
> 
> 
> 
> [![Well](https://telegra.ph/file/32681b28f0c194a796edc.png)](https://telegra.ph/file/fa078713c5e6236075f3b.mp4)

**Reply by Ray-Senpai07** [reply]
> > ### Select Topic Area
> > Product Feedback
> > ### Body
> > Dear GitHub Team,
> > I'm a phone user
> > I wanted to provide some feedback on the recent changes made to the GitHub code view. While I appreciate the updates, I have encountered a few issues that have affected my coding experience negatively.
> > The new code view is laggy, glitchy, and makes it hard to delete new lines or copy-paste code. Personally, I preferred the previous version as it was smoother and more user-friendly.
> > Please consider the feedback from developers like myself who are facing difficulties with the new code view. Enhancing the user experience will help maintain GitHub as a beloved platform for developers worldwide.
> > [![Well](https://camo.githubusercontent.com/e41dab59eee0beaf7f0d45d92c6a5e019f24455c5d0e5b728b06cf6512075e53/68747470733a2f2f74656c656772612e70682f66696c652f3332363831623238663063313934613739366564632e706e67)](https://telegra.ph/file/fa078713c5e6236075f3b.mp4)

Man its a legit glich

**Reply by Billywildhig** [reply]
> > > ### Select Topic Area
> 
> > > Product Feedback
> 
> > > ### Body
> 
> > > Dear GitHub Team,
> 
> > > I'm a phone user
> 
> > > I wanted to provide some feedback on the recent changes made to the GitHub code view. While I appreciate the updates, I have encountered a few issues that have affected my coding experience negatively.
> 
> > > The new code view is laggy, glitchy, and makes it hard to delete new lines or copy-paste code. Personally, I preferred the previous version as it was smoother and more user-friendly.
> 
> > > Please consider the feedback from developers like myself who are facing difficulties with the new code view. Enhancing the user experience will help maintain GitHub as a beloved platform for developers worldwide.
> 
> > > [![Well](https://camo.githubusercontent.com/e41dab59eee0beaf7f0d45d92c6a5e019f24455c5d0e5b728b06cf6512075e53/68747470733a2f2f74656c656772612e70682f66696c652f3332363831623238663063313934613739366564632e706e67)](https://telegra.ph/file/fa078713c5e6236075f3b.mp4)
> 
> 
> 
> Man its a legit glich

**Comment by Billywildhig** [comment]
[]()

**Reply by AliAryanTech** [reply]
What the?

**Reply by krystian3w** [reply]
He pasted empty link:

![image](https://github.com/community/community/assets/35370833/1f125168-2cce-4bd1-830e-b1ac595c8a85)

?

**Comment by AliAryanTech** [comment]
I apologize in advance if I am tagging the wrong person, but I am trying to get the attention of the individuals responsible for the new code interface feature on Github.

Thank you for your time and assistance.
@queenofcorgis @natalierjackson @otter-computer @martinwoodward

**Comment by Cbodfield** [comment, marked as answer]
I apologize for the friction on mobile editing, @AliAryanTech. I'm actively working on fixing that problem and enhancing that experience.

**Reply by AliAryanTech** [reply]
Okay, thanks for your response

**Reply by AliAryanTech** [reply]
Can you please allow us to use old interface, I need to do my project but I can't coz it's lagging so until you guys working on it please allow us to disable this new interface, we need to complete our projects on time.

**Reply by Cbodfield** [reply]
@AliAryanTech I'm wrapping up the fixes and the improved edit experience. It may not be wildly available for another week. However, if you'd like I can opt you in early as an early tester?

**Reply by AliAryanTech** [reply]
> @AliAryanTech I'm wrapping up the fixes and the improved edit experience. It may not be wildly available for another week. However, if you'd like I can opt you in early as an early tester?

Sure, I really appreciate it

**Reply by Cbodfield** [reply]
Added! You may need to refresh to get the new changes while editing. 

Please `@` me directly in this discussion with any new issues you find or general feedback.

**Reply by AliAryanTech** [reply]
Got it! I'll keep you updated on any issues or feedback. Thank you for including me as a beta tester!

**Reply by Cbodfield** [reply]
Hey @AliAryanTech 👋  just checking in to see how things are going with the beta mobile edit changes

**Reply by AliAryanTech** [reply]
Oh- It's good enough, now I can easily do my projects it's quite stable, but I guess it need a bit improvement in viewing code, I mean it seems like emoji glitch 


https://github.com/community/community/assets/85696485/bfe70413-c4fa-4b30-b86f-53961f165117

**Reply by Cbodfield** [reply]
Glad to hear that things are working better for you on the edit side of things!

I'll pass that emoji bug video on to the rest of the team to look into. As a mobile-only user, are there any other pain points or areas of improvement with viewing code that I can pass along?

**Reply by AliAryanTech** [reply]
Oh thanks, if I find anything which needs to be improve I'll inform you here.

For now I think it'll be good if users have an option of wrap on/off while viewing commit changes

![Screenshot_20230601-204947](https://github.com/community/community/assets/85696485/2a9e026c-83c2-4b34-a3c5-0d0dd6e67565)

**Reply by AliAryanTech** [reply]
Hey, @Cbodfield 👋
I was just enabling two-factor authentication (2FA) by using Setup SMS authentication but the Country Pakistan isn't available. I was thinking if you can do something about this?

**Reply by Cbodfield** [reply]
@AliAryanTech That is something I can't really help much with. This is the [list of countries](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/countries-where-sms-authentication-is-supported) that we currently have SMS support for. You may need to setup 2FA with a [different method](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/configuring-two-factor-authentication). Our [support site](https://support.github.com/) may have additional info.

**Reply by AliAryanTech** [reply]
Oh- okay thanks

**Reply by jfioasd** [reply]
@Cbodfield Apparently, multi-line editing is an issue as well. When I try to copy several lines, the line numbers get inside.

When I try to delete a line using backspace, it doesn't work and I had to cut the line from the code (which is annoying because usually I had to cut some words out of the editing space for it to be carried out successfully).

**Reply by Cbodfield** [reply]
@jfioasd I apologize for the friction! The changes that are coming soon should solve those problems. Would you like me to add you as a beta tester to confirm?

**Reply by AliAryanTech** [reply]
Hi @Cbodfield, it's was like usually day I was coding some stuff, and unfortunately I'm facing some friction while writing, and sometimes the code disappear for a sec and when I move the tab [code box] it appears, I don't know what's wrong!?

Am I not beta user anymore??

**Reply by Cbodfield** [reply]
Hey @AliAryanTech - sorry to hear that you're having more friction. The changes have been rolled out to everyone so this may be a new bug. The only exception is that a handful of file types have not been fully migrated yet.

Do you have more info on the file you were editing or how to replicate the problem?

**Reply by AliAryanTech** [reply]
Actually, It's not happening now
but there's something which is not good
I can't see whole search bar, nor movable 

![Screenshot](https://github.com/community/community/assets/85696485/9f8a4b17-2095-49f9-82ab-10ca165eb347)

**Reply by Cbodfield** [reply]
Thanks for the heads up @AliAryanTech. I'll pass that bug along to the right team to get it fixed.

**Reply by AliAryanTech** [reply]
Hi @Cbodfield, 👋
Did you guys fixed the search engine bug?
Anyway, Can you fix this? I'm unable to move file to another folder

https://github.com/community/community/assets/85696485/1974b21c-03e7-4be3-b49a-c8e4a5d26d35

**Reply by Cbodfield** [reply]
@AliAryanTech I'm not sure the status on the search dialog fix. I'll get the change folder bug on our radar for fixing. Thanks again for the great feedback!

**Reply by AliAryanTech** [reply]
Hi @Cbodfield 
Long time no see, I tagged you to report about the delete button, I mean the "commit changes" button we use for that action. Can you fix it?

![Screenshot_20231018-234410](https://github.com/community/community/assets/85696485/945cb71e-9b80-4a98-927a-ba2981f9679f)

**Reply by Cbodfield** [reply]
Hey @AliAryanTech! Thanks for reaching out and reporting that issue. I've passed it along to the team to fix.

**Reply by AliAryanTech** [reply]
Hey @Cbodfield 
Don't you think the buttons CSS isn't correct?
"cancel changes" and "commit changes" should be together!?

![Screenshot_20231022-212257](https://github.com/community/community/assets/85696485/86f90d1e-15b2-4707-98a7-c37b5da01d20)

I think it should be like this
![Screenshot_20231022-223852](https://github.com/community/community/assets/85696485/ec1e4065-9daf-431a-9726-2652f11a2b19)

Also one question; Is it possible to set indent size as 4 by default?
Whenever I open a file the indent size is 2 as default

**Reply by Cbodfield** [reply]
I'll pass along that button position feedback. 👍 

As for the indent size - there is not currently a user-level setting for that. What you can do though is add an `.editorconfig` file to the root of your repository. 

```
# This file is the top-most EditorConfig file
root = true

# All Files
[*]
indent_size = 4
```

**Reply by AliAryanTech** [reply]
Hi @Cbodfield, it's a report of GitHub interface update (Need to fix CSS)

Image:
![Screenshot_20231207-163625](https://github.com/community/community/assets/85696485/cc7597ee-6ac4-4f91-8470-ffd344aad5d0)

I think this one too

![Screenshot_20231207-163608](https://github.com/community/community/assets/85696485/39caa2c0-227c-4340-a332-39ac40bda378)

Also, it's a off topic question;
Do you know some TikTok Developer's?
I reported a account many times in different email's of TikTok, but still no response.

**Comment by lq920320** [comment]
In this view, anybody can return back to the before version? I can't read markdown with image in this code view, or is there some way that 
 how to preview markdown file?

**Reply by lq920320** [reply]
![image](https://github.com/community/community/assets/18696549/12740f99-b2fa-483a-a190-c0f9f2dab988)

**Reply by krystian3w** [reply]
For `*.md` files exists a "preview" button:

> ![image](https://github.com/community/community/assets/35370833/2aa43aef-71af-4ef9-896a-06700b214513) |
> --- |

> ![image](https://github.com/community/community/assets/35370833/daab744a-e21d-44b0-8fd7-6dda5fb6a403) |
> --- |

maybe  with `.gitattributes` possible override language of file to markdown with "preview" button:

```gitattributes
*.txt linguist-language=markdown linguist-detectable
# *.txt linguist-language=Markdown linguist-detectable
# *.txt linguist-language=MarkDown linguist-detectable
```

The effect may be visible after a few days, If nothing changes after a week, you may need to consider changing the file format from wrong with only highlighter to `*.md`.

The gray link indicates that linguist probably did not recognize the file as markdown. Correct have underline:

> ![image](https://github.com/community/community/assets/35370833/9017a594-f2ad-4883-8d98-c1ea87a46dc5) |
> --- |

**Reply by lq920320** [reply]
OK, I see. The code repository I viewed had the problem. Thanks!
